### PR TITLE
day 6 solutions

### PR DESCRIPTION
**Pull request by rajivkr**

### CODE REVIEW

File: js-exercises/alternating-characters/alternatingCharacters.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-plusplus */
+const alternatingCharacters = (arrOfInputStrings = []) =>
+  // eslint-disable-next-line implicit-arrow-linebreak
+  arrOfInputStrings.map(input => {
+    let minimumDeletionCount = 0;
+    const everyInputArr = input.split('');
```
Comment (sam201994) [inline js-exercises/alternating-characters/alternatingCharacters.js:6]: you could extract out the logic for a single string in a different function

Comment (sam201994) [inline js-exercises/alternating-characters/alternatingCharacters.js:6]: `everyInputArr` does not sound like a good variable name. It is a little confusing

File: js-exercises/user-check-password/user.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function User(username, password) {
+  Object.assign(this, { username, password });
+}
+
+// eslint-disable-next-line
+User.prototype.checkPassword = function(passToCompare) {
```
Comment (sam201994) [inline js-exercises/user-check-password/user.js:6]: maybe `passwordToCompare` could be better

File: js-exercises/user-check-password/user.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+function User(username, password) {
+  Object.assign(this, { username, password });
```
Comment (sam201994) [inline js-exercises/user-check-password/user.js:2]: just curious why have you used Object.assign here? is it needed? I don't know myself. As far as I know we can just do this.username = username

File: js-exercises/alternating-characters/alternatingCharacters.js (hunk 1 above)
Comment (rajivkr) [reply js-exercises/alternating-characters/alternatingCharacters.js:6]: yep, inputArr sounds better now :)

File: js-exercises/user-check-password/user.js (hunk 3 above)
Comment (rajivkr) [reply js-exercises/user-check-password/user.js:2]: Both are almost similar, Object.assign is generally used in the place where we need to copy only the enumerable objects, but yes, in this place, this.username serves the purpose.

**Pull request merged**